test/js/setupTests.js

```diff
@@ -109,3 +109,6 @@ expect.extend({
       }));
   }
 });
+
+// For some reason, the process.on('exit') handler does not get called in jest specs even when the process exits. To remedy this, we emit a custom event.
+afterAll(() => process.emit("exit"));
```

Comment:
Check out globalTeardown for this. afterAll gets called a couple of times unfortunately.

Reply:
Looks like I dont need to do this anymore since we are using the global async parser server, good catch